editor/src/clj/editor/command_requests.clj

```diff
@@ -0,0 +1,270 @@
+;; Copyright 2020-2023 The Defold Foundation
+;; Copyright 2014-2020 King
+;; Copyright 2009-2014 Ragnar Svensson, Christian Murray
+;; Licensed under the Defold License version 1.0 (the "License"); you may not use
+;; this file except in compliance with the License.
+;;
+;; You may obtain a copy of the License, together with FAQs at
+;; https://www.defold.com/license
+;;
+;; Unless required by applicable law or agreed to in writing, software distributed
+;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
+;; specific language governing permissions and limitations under the License.
+
+(ns editor.command-requests
+  (:require [clojure.data.json :as json]
+            [clojure.string :as string]
+            [dynamo.graph :as g]
+            [editor.disk :as disk]
+            [editor.ui :as ui]
+            [service.log :as log])
+  (:import [java.nio.charset StandardCharsets]))
+
+(set! *warn-on-reflection* true)
+
+(defonce ^:const url-prefix "/command")
+
+(defonce ^:private supported-commands
+  ;; Notable exclusions:
+  ;; :save-all, :quit, anything that would open a modal dialog.
+  {:asset-portal
+   {:ui-handler :asset-portal
+    :help "Open the Asset Portal in a web browser."}
+
+   :async-reload
+   {:ui-handler :async-reload
+    :help "Reload all modified files from disk."}
```

Comment:
I think we should not expose this command and make reloading an implementation detail to not push this complexity onto our users.

Reply:
I did not intend for this to be necessary for users to invoke this manually, but was imagining a use case where you had the editor open in a scene view on a second monitor while you ran a tool that generated content. That way, the tool could ask the editor to refresh the view without you having to tab into the editor.

Although TBH I'm not sure we refresh the scene view while the editor is in the background at the moment. 😅

Reply:
I decided to remove the reload command for now, since it is only available if you're have the "Load External Changes on App Focus" preference disabled, which feels wrong.